src/natcap/invest/validation.py

```diff
@@ -538,15 +560,43 @@ def check_csv(filepath, required_fields=None, excel_ok=False):
             return ("File could not be opened as a CSV. "
                     "File must be encoded as a UTF-8 CSV.")
 
-    if required_fields:
-        dataframe.columns = dataframe.columns.str.strip()
-        fields_in_table = set([name.upper() for name in dataframe.columns])
-        missing_fields = (
-            set(field.upper() for field in required_fields) - fields_in_table)
+    # assume that at most one of `rows` and `columns` is defined
+    if columns:
+        headers = [str(name).strip() for name in dataframe.iloc[0]]
+        return check_headers(get_headers_to_validate(columns), headers, 'column')
+    elif rows:
+        headers = [str(name).strip() for name in dataframe.iloc[:, 0]]
+        return check_headers(get_headers_to_validate(rows), headers, 'row')
+
+
+def check_headers(expected_headers, actual_headers, header_name='header'):
+    """Validate that table headers (rows/columns) match a list of patterns.
+
+    - Each pattern should match at least one header.
+    - No header should be matched by more than one pattern.
+    - Each header that's matched by any pattern should not be duplicated.
+    - Headers are allowed to not match any pattern.
+    - Headers are converted to lowercase before matching.
+
+    Args:
+        patterns (list[str]): A list of patterns expected to match header(s).
+            Each pattern should be parse-able as regex and will be anchored to
+            the header start and end.
+        headers (list[str]): A list of headers to validate.
+        header_name (str): A string to use in the error message to refer to the
+            header (typically one of 'column', 'row', 'field')
```

Comment:
I think this docstring is out of sync with the function definition now.

Reply:
fixed in dc72cfc1